src/Hook.php

```diff
@@ -0,0 +1,240 @@
+<?php
+
+namespace Utopia;
+
+class Hook {
+
+    /**
+     * Description
+     *
+     * @var string
+     */
+    protected string $desc = '';
+
+    /**
+     * Parameters
+     *
+     * List of route params names and validators
+     *
+     * @var array
+     */
+    protected array $params = [];
+
+    /**
+     * Group
+     *
+     * @var array
+     */
+    protected array $groups = [];
+
+    /**
+     * Action Callback
+     *
+     * @var callable
+     */
+    protected $action;
+
+    /**
+     * Injections
+     *
+     * List of route required injections for action callback
+     *
+     * @var array
+     */
+    protected array $injections = [];
+
+    /**
+     * Add Description
+     *
+     * @param string $desc
+     * @return static
+     */
+    public function desc(string $desc): static
+    {
+        $this->desc = $desc;
+
+        return $this;
+    }
+
+    /**
+     * Get Description
+     *
+     * @return string
+     */
+    public function getDesc(): string
+    {
+        return $this->desc;
+    }
+
+    /**
+     * Add Group
+     *
+     * @param array $groups
+     * @return static
+     */
+    public function groups(array $groups): static
```

Comment:
Should the return type be `static` or `self` ? Since this is not a static method, I think it might be better to use `self`

Reply:
@christyjacob4 please refer to this
https://php.watch/versions/8.0/static-return-type

Reply:
This is so that IDE support in child classes that extend this is proper with return type.